### PR TITLE
[doc][core] Fix doc for runtime-env-auth

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?

Fix the error introduced by https://github.com/ray-project/ray/commit/6a13b094585bd85674825088779077dc1d03104c

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by shrekris-anyscale — APPROVED** [review]
Thanks for making this!

**Review by angelinalg — APPROVED** [review]

**Comment by rickyyx** [comment]
This is still not working yet i believe.

**Comment by rickyyx** [comment]
Ok it's working now: https://anyscale-ray--36421.com.readthedocs.build/en/36421/ray-core/handling-dependencies.html

**Comment by rickyyx** [comment]
So apparently `_` and `-` makes a difference.

File: doc/source/ray-core/runtime-env-auth.md
Comment (matthewdeng) [inline doc/source/ray-core/runtime-env-auth.md:1]: I saw this warning in the job.
```
/ray/doc/source/ray-core/runtime-env-auth.md: WARNING: document isn't included in any toctree
```

Looks like it is excluded here:
https://github.com/ray-project/ray/blob/319080c9350b3634c40aa63fa3ca8e02365bb161/doc/source/conf.py#L191

Could this be what's actually causing the issue?

Comment (rickyyx) [reply doc/source/ray-core/runtime-env-auth.md:1]: hmmm, this is weird....

Comment (matthewdeng) [reply doc/source/ray-core/runtime-env-auth.md:1]: Should this page be included in the table of contents somewhere? I feel that if you add it and remove this line in `conf.py`, the change currently in this PR may not be needed.

Comment (rickyyx) [reply doc/source/ray-core/runtime-env-auth.md:1]: i think yeah, maybe we should add to a TOC somewhere.

Comment (rickyyx) [reply doc/source/ray-core/runtime-env-auth.md:1]: let me try.

**Review by matthewdeng — APPROVED** [review]
<div><br/>via <a href="https://giphy.com/thefastsaga/">The Fast Saga</a> on <a href="https://giphy.com/gifs/thefastsaga-fast-and-furious-tokyo-drift-toz7qXlLyHy9n8KfKO">GIPHY</a></div>

**Pull request merged**